wandb/apis/public.py

```diff
@@ -1916,22 +1916,30 @@ def file(self, name):
     def upload_file(self, path, root="."):
         """
         Arguments:
-            path (str): name of file to upload.
-            root (str): the root path to save the file relative to.  i.e.
-                If you want to have the file saved in the run as "my_dir/file.txt"
-                and you're currently in "my_dir" you would set root to "../"
+            path (str): local path to the file to upload.
+            root (str): the root path to save the file in wandb.
```

Comment:
> As of https://github.com/wandb/wandb/pull/3924 after exploring wandb's Public API, I saw that the root parameter in wandb.apis.public.File.upload_file didn't handle file paths just directories

I believe this is the intention. I see the pain point you're addressing in this PR, but it feels like overloading the meaning of parameters here will not result in a cleaner API.